cmd/k8s-operator/ingress-for-pg.go

```diff
@@ -473,16 +486,22 @@ func (r *HAIngressReconciler) maybeCleanup(ctx context.Context, hostname string,
 	if err != nil {
 		return false, fmt.Errorf("error deleting VIPService: %w", err)
 	}
+
+	// 3. Clean up any cluster resources
+	if err := r.cleanupCertResources(ctx, pg, cfg.Services[serviceName]); err != nil {
```

Comment:
Should we move this to after the nil check just below if it could be nil?

Reply:
I'll add logic to not require serve config to calculate the domain name. We should be able to clean up Ingress resources even when ProxyGroup was deleted first

Reply:
Nice, that's way better